src/core/Akka.API.Tests/CoreAPISpec.ApproveClusterSharding.approved.txt

```diff
@@ -72,19 +72,29 @@ namespace Akka.Cluster.Sharding
     {
         public readonly System.Collections.Immutable.IImmutableDictionary<Akka.Actor.Address, Akka.Cluster.Sharding.ShardRegionStats> Regions;
         public ClusterShardingStats(System.Collections.Immutable.IImmutableDictionary<Akka.Actor.Address, Akka.Cluster.Sharding.ShardRegionStats> regions) { }
-        public bool Equals(Akka.Cluster.Sharding.ClusterShardingStats other) { }
         public override bool Equals(object obj) { }
+        public bool Equals(Akka.Cluster.Sharding.ClusterShardingStats other) { }
         public override int GetHashCode() { }
+        public override string ToString() { }
     }
-    public sealed class CurrentRegions
+    public sealed class CurrentRegions : Akka.Cluster.Sharding.IClusterShardingSerializable, System.IEquatable<Akka.Cluster.Sharding.CurrentRegions>
     {
         public readonly System.Collections.Immutable.IImmutableSet<Akka.Actor.Address> Regions;
         public CurrentRegions(System.Collections.Immutable.IImmutableSet<Akka.Actor.Address> regions) { }
+        public override bool Equals(object obj) { }
+        public bool Equals(Akka.Cluster.Sharding.CurrentRegions other) { }
+        public override int GetHashCode() { }
+        public override string ToString() { }
     }
-    public sealed class CurrentShardRegionState
+    public sealed class CurrentShardRegionState : Akka.Cluster.Sharding.IClusterShardingSerializable, System.IEquatable<Akka.Cluster.Sharding.CurrentShardRegionState>
     {
+        public readonly System.Collections.Immutable.IImmutableSet<string> Failed;
         public readonly System.Collections.Immutable.IImmutableSet<Akka.Cluster.Sharding.ShardState> Shards;
-        public CurrentShardRegionState(System.Collections.Immutable.IImmutableSet<Akka.Cluster.Sharding.ShardState> shards) { }
+        public CurrentShardRegionState(System.Collections.Immutable.IImmutableSet<Akka.Cluster.Sharding.ShardState> shards, System.Collections.Immutable.IImmutableSet<string> failed) { }
```

Comment:
Avoid changing an existing public API method signature, this should be made into a function overload.

Reply:
added